driver/modern_bpf/helpers/base/push_data.h

```diff
@@ -89,6 +89,9 @@ enum read_memory
 /* Paramater maximum size */
 #define MAX_PARAM_SIZE MAX_EVENT_SIZE - 1
 
+/* ARGS or ENV maximum size */
+#define ARGS_ENV_SIZE_MAX 4096
```

Comment:
not completely understood why we need this :thinking:

Reply:
`SAFE_ACCESS` should already protect us from the worst case

Reply:
>It has been part of the old probe and it may be less for protection and rather more for you don't wanna ever log super super long unbounded strings in your logs.

That's a good point for this reason we use approaches like the `snaplen` to let users understand the max dimension they want to send, you can see an example here https://github.com/falcosecurity/libs/blob/0c280ca6847d7fbb616f152bb6cffd5b4d74452d/driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/sendto.bpf.c#L89-L98

The idea of this  `snaplen` concept is to limit at an higher level so you can chose when to limit rather then implementing it directly at low level

For what concerns the old probe we use this same concept of `snaplen` all the others checks are to please the verifier AFAIK :thinking:

Reply:
Those are the two relevant max bounds from old probe

```
#define ARGS_ENV_SIZE_MAX 4096
#define MAX_PATH_LENGTH 4096
```

These values are battle proven, so wouldn't go higher before testing modern_bpf in the field.

Probably have a few options here and we can go step-wise, no need to change all now.

- Could start with uniformly enforcing bounds to existing `MAX_EVENT_SIZE`, if I can say so, the value of `2^16` appears extremely large, I think `2^12` is reasonable and enough and as mentioned battle proven. If we go this route can change this quickly and reuse the existing macro.
- If we need more granular upper bounds can introduce a new enum, similarly to the `read_memory`. For example, for the cmd args I can see that maybe `2^13` could be a good choice for an enforced upper bound. Would however perform all bound checks and selecting the corresponding upper bound here and not not in the fillers. In addition, fillers can always also enforce a lower value than the max enforced bound.

These bound checks are not unique to Falco or libs, seeing them everywhere and I would be very hesitant deploying anything to production without them.

Reply:
Sorry typo meant the `MAX_PARAM_SIZE` you also use in `SAFE_ACCESS`

Reply:
uhm I got your point maybe `MAX_PARAM_SIZE` is big enough, it's the same value we use in the old probe but it could be dangerous...

```c
/* Event maximum size */
#define MAX_EVENT_SIZE 64 * 1024

/* Paramater maximum size */
#define MAX_PARAM_SIZE MAX_EVENT_SIZE - 1
````

BTW putting a hard-coded value here doesn't seem the right solution, or better let's say it could be the right solution if we were in production and we have to rush, but this is not the case :rofl: let's take our time to think of something more dynamic or configurable

Reply:
I've added some dedicated checks for `execve` here https://github.com/falcosecurity/libs/pull/648/files#diff-23e1e00c90562960e7e4110fcc5c62cd89d2ea8a8df3198d7c04512084c7c378R52-R54 :)